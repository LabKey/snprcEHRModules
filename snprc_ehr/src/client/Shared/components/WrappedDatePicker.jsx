/* eslint max-classes-per-file: ["error", 2] */

import React from 'react'
import DatePicker from 'react-datepicker'

export default class WrappedDatePicker extends React.PureComponent {
    render() {
        return (
          <>
            { this.props.label && <label className="field-label">{this.props.label}</label> }
            <DatePicker
              customInput={ <DatePickerInput /> }
              dateFormat={ this.props.dateFormat }
              disabled={ this.props.disabled }
              disabledKeyboardNavigation={ this.props.disabledKeyboardNavigation }
              dropdownMode="select"
              id="date-picker"
              maxDate={ this.props.maxDate }
              onChange={ this.props.onChange }
              onChangeRaw={ this.props.onChangeRaw }
              onSelect={ this.props.onSelect }
              popperPlacement="bottom"
              readOnly={ this.props.readOnly }
              selected={ this.props.selected }
              showMonthDropdown
              showTimeSelect={ this.props.showTimeSelect }
              showYearDropdown
              timeFormat={ this.props.timeFormat }
              timeIntervals={ this.props.timeIntervals }
              todayButton={ this.props.todayButton }
            />
          </>
        )
    }
}

class DatePickerInput extends React.PureComponent {
    render() {
        return (
          <div className="datepicker-input-wrapper">
            <input
              className="datepicker-input"
              disabled={ this.props.disabled }
              onChange={ this.props.onChange }
              onClick={ this.props.onClick }
              type="text"
              value={ this.props.value }
            />
            <i aria-hidden="true" className="fa fa-calendar" onClick={ this.props.onClick } />
          </div>
        )
    }
}
