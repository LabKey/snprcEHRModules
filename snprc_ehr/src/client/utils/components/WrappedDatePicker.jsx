import React from 'react';
import DatePicker from 'react-datepicker';

export default class WrappedDatePicker extends React.Component {
    render() {
      return (
            <>
                { this.props.label && <label className="field-label" >{this.props.label}</label> }
                <DatePicker
                    id="date-picker"
                    todayButton={this.props.todayButton}
                    showTimeSelect={this.props.showTimeSelect}
                    timeFormat={this.props.timeFormat}
                    timeIntervals={this.props.timeIntervals}
                    dateFormat={this.props.dateFormat}
                    maxDate={this.props.maxDate}
                    customInput={<DatePickerInput/>}
                    selected={this.props.selected}
                    onSelect={this.props.onSelect}
                    onChange={this.props.onChange}
                    onChangeRaw={this.props.onChangeRaw}
                    disabledKeyboardNavigation={this.props.disabledKeyboardNavigation}
                    readOnly={this.props.readOnly}
                    disabled={this.props.disabled}
                    popperPlacement="bottom"
                    showMonthDropdown
                    showYearDropdown
                    dropdownMode="select"
                />
            </>
      )
    }
}

class DatePickerInput extends React.Component {
    render() {
      return (
        <div className="datepicker-input-wrapper" >
          <input 
              onClick={this.props.onClick} 
              onChange={this.props.onChange}
              className="datepicker-input" 
              value={this.props.value} 
              type="text" 
              disabled={this.props.disabled}
            />
          <i onClick={this.props.onClick} aria-hidden="true" className="fa fa-calendar"></i>
        </div>
      )
    }
  }