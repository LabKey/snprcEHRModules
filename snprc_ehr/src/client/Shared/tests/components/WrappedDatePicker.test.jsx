import React from 'react'
import { shallow } from 'enzyme'
import moment from 'moment'
import WrappedDatePicker from '../../components/WrappedDatePicker'

test('Should render the WrappedDatePicker', () => {
    const wrapper = shallow(
      <WrappedDatePicker
        dateFormat="Pp"
        disabled={ false }
        id="acquisition-datepicker"
        label="Acquisition Date"
        maxDate={ moment().toDate() }
        onChange={ () => {} }
        onSelect={ () => {} }
        selected={ moment().toDate() }
        showTimeSelect
        timeFormat="p"
        timeIntervals={ 30 }
        todayButton="Today"
      />
    )
    expect(wrapper).toMatchSnapshot()
})
