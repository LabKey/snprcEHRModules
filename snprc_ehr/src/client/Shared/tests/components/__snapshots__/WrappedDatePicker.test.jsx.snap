// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the WrappedDatePicker 1`] = `
<Fragment>
  <label
    className="field-label"
  >
    Acquisition Date
  </label>
  <o
    allowSameDay={false}
    customInput={<DatePickerInput />}
    customTimeInput={null}
    dateFormat="Pp"
    dateFormatCalendar="LLLL yyyy"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="select"
    enableTabLoop={true}
    excludeScrollbar={true}
    id="date-picker"
    inlineFocusSelectedMonth={false}
    maxDate={1970-01-01T00:00:00.000Z}
    monthsShown={1}
    nextMonthButtonLabel="Next Month"
    nextYearButtonLabel="Next Year"
    onBlur={[Function]}
    onCalendarClose={[Function]}
    onCalendarOpen={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onInputClick={[Function]}
    onInputError={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    popperPlacement="bottom"
    preventOpenOnFocus={false}
    previousMonthButtonLabel="Previous Month"
    previousYearButtonLabel="Previous Year"
    readOnly={false}
    renderDayContents={[Function]}
    selected={1970-01-01T00:00:00.000Z}
    shouldCloseOnSelect={true}
    showFullMonthYearPicker={false}
    showMonthDropdown={true}
    showMonthYearPicker={false}
    showPopperArrow={true}
    showPreviousMonths={false}
    showQuarterYearPicker={false}
    showTimeInput={false}
    showTimeSelect={true}
    showYearDropdown={true}
    showYearPicker={false}
    strictParsing={false}
    timeCaption="Time"
    timeFormat="p"
    timeInputLabel="Time"
    timeIntervals={30}
    todayButton="Today"
    withPortal={false}
  />
</Fragment>
`;
